Service/Manager.php

```diff
@@ -265,6 +283,8 @@ public function persist($document)
         $documentArray = $this->converter->convertToArray($document);
         $type = $this->getMetadataCollector()->getDocumentType(get_class($document));
 
+        $this->eventDispatcher->dispatch(Events::PERSIST, new PersistEvent($document));
```

Comment:
I think this even is not necessary. To modify all documents we can in pre_commit event.